internal/compiler-bridge/src/main/scala/xsbt/API.scala

```diff
@@ -22,12 +23,31 @@ final class API(val global: CallbackGlobal) extends Compat with GlobalHelpers {
   class ApiPhase(prev: Phase) extends GlobalPhase(prev) {
     override def description = "Extracts the public API from source files."
     def name = API.name
+
     override def run(): Unit = {
       val start = System.currentTimeMillis
       super.run()
       callback.apiPhaseCompleted()
       val stop = System.currentTimeMillis
       debuglog("API phase took : " + ((stop - start) / 1000.0) + " s")
+
+      if (captureUnits) units = currentRun.units.map(u => (u, u.body)).toList
+    }
+
+    // Hack to make this phase independentally benchmarkable
+    private val captureUnits: Boolean = java.lang.Boolean.getBoolean(API.ApiExtractBenchmarkModeKey)
+    private var units: List[(CompilationUnit, Tree)] = Nil
```

Comment:
Could this variable avoid GC collection for the tree references? I'm not sure how the life management of phases is done in scalac, hence the question.

Reply:
Firstly, it is only populated other than in this special test mode.

In general, the `Phase`-s have the same lifetime as the `Run` that assembles them. This is a bit longer than it needs to be as `Global.curRun` is not nulled out, it is only overwrritten on the creation of the next `Run`.

In the case of sbt/Zinc, the `Global` instance itself it discarded after compilation. So the eager nulling of `Global.curRun` would only be beneficial for long-lived `Global`-s, e.g in Scala IDE or the REPL.